docs/dev/testing.md

```diff
@@ -0,0 +1,76 @@
+# Assisted Installer Testing
+
+The assisted-installer tests are divided into 3 categories:
+
+* **Unit tests** - Focused on a module/function level while other modules are mocked.
+Unit tests are located next to a module named `module_name_test.go`
+
+* **Subsystem tests** - Focused on the component while mocking other component. (e.g. agent/installer results).
+
+* **System tests** (a.k.a e2e) - Running full flows with all components.
+The e2e tests are divided into u/s (upstream) basic workflows on [assisted-test-infra](https://github.com/openshift/assisted-test-infra/tree/master/discovery-infra/tests) and d/s (downstream) extended regression tests maintained by both DEV and QE teams on [kni-assisted-installer-auto](https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/tree/master/api_tests).
+
+## Repository CI
+
+| [Jenkins]((http://assisted-jenkins.usersys.redhat.com)) | [Prow](https://prow.ci.openshift.org) |
+|---|---|
+| Local for Assisted ecosystem | Company-wide |
```

Comment:
What do the rows mean in this table?

Reply:
Comparison between the two instances

Reply:
That's the columns, it's not clear what the rows mean. The entire table is kinda of confusing, maybe it would be better to have it as text instead